File: src/queue/Queue.java
package queue;

public class Queue<T> {
    private T[] items;
    private int length;

    public Queue() {
        this.items = (T[]) new Object[10];
        this.length = 0;
    }

    public void enqueue(T item) {

        this.aumentarCapacidade();
        this.items[this.length] = item;
        this.length++;
    }

    private void aumentarCapacidade() {

        if (this.length == this.items.length) {

            T[] newItemsList = (T[]) new Object[this.items.length *2];

            for(int i = 0; i < this.items.length; i++){
                newItemsList[i] = this.items[i];
            }
            this.items = newItemsList;

        }
    }

    public T dequeue() {
        if(this.isEmpty()){
            return null;
        }
        int POSITTION=0;

        T deletedItem=this.items[POSITTION];

        for (int i = POSITTION; i < this.length; i++) {
                this.items[i] = this.items[i+1];

        }

        this.length--;
        return deletedItem;

    }

    public boolean isEmpty() {
        return this.length == 0;
    }

    public T peek() {
        if(this.isEmpty()){
            return null;
        }
        return this.items[0];
    }

    public int size() {
        return this.length;
    }

    public void clear() {
        this.items = (T[]) new Object[10];
        this.length = 0;
    }

    @Override
    public String toString() {

        StringBuilder stringBuilder = new StringBuilder();
        stringBuilder.append("[");

        for (int i = 0; i < this.length-1; i++) {
            stringBuilder.append(this.items[i]);
            stringBuilder.append(", ");
        }

        if (this.length > 0) {
            stringBuilder.append(this.items[this.length-1]);
        }

        stringBuilder.append("]");
        return stringBuilder.toString();
    }
}

File: em-python/Queue.py
class Queue:
    def __init__(self):
        self.items = [None] * 10
        self.length = 0

    def enqueue(self, item):
        self.resize()
        self.items[self.length] = item
        self.length += 1

    def resize(self):
        if self.length == len(self.items):
            new_items = [None] * (len(self.items) * 2)

            for i in range(len(self.items)):
                new_items[i] = self.items[i]

            self.items = new_items

    def dequeue(self):
        if self.is_empty():
            return None

        position = 0
        deleted_item = self.items[position]

        for i in range(position, self.length - 1):
            self.items[i] = self.items[i + 1]

        self.items[self.length - 1] = None
        self.length -= 1

        return deleted_item

    def peek(self):
        if self.is_empty():
            return None
        return self.items[0]

    def is_empty(self):
        return self.length == 0

    def size(self):
        return self.length

    def clear(self):
        self.items = [None] * 10
        self.length = 0

    def __str__(self):
        result = "["
        for i in range(self.length - 1):
            result += str(self.items[i]) + ", "

        if self.length > 0:
            result += str(self.items[self.length - 1])

        result += "]"
        return result